### PR TITLE
Add configure-db script

### PR DESCRIPTION
**Pull request by fviale**
Also:
 - improved error logging in LoadPackage.groovy
 - improved encrypted password handling in HSQLDB configuration
 - improved configure-http output

### CODE REVIEW

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+
+    public static final String URL_OPTION = "r";
+
+    public static final String URL_OPTION_NAME = "url";
+
+    public static final String USERNAME_OPTION = "u";
+
+    public static final String USERNAME_OPTION_NAME = "username";
+
+    public static final String PASSWORD_OPTION = "p";
```
Comment (mboussaa) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java:69]: -P for password
and
-p for port
isn't better than -p and -t  ?

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,686 @@
...
+        if (cmd.hasOption(MYSQL_TIMEZONE_OPTION_NAME)) {
+            if (url != null) {
+                exitWithErrorMessage("When --" + URL_OPTION_NAME + " is provided, the --" + MYSQL_TIMEZONE_OPTION_NAME +
+                                     " option cannot be used", null, null);
+            }
+            timezone = cmd.getOptionValue(MYSQL_TIMEZONE_OPTION_NAME).trim();
+        } else if (url == null) {
+            timezone = "UTC";
+        }
+
```
Comment (mboussaa) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java:366]: either add line break everywhere between Ifs or not

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java (hunk 1 above)
Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java:69]: I forgot that it was possible to use uppercase latter. I would use the opposite though, as I intend to do the same for hostname

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,686 @@
...
+                                "Password will always be encrypted when stored in configuration files.");
+        opt.setRequired(false);
+        opt.setArgName(PASSWORD_OPTION_NAME.toUpperCase());
+        opt.setArgs(1);
+        options.addOption(opt);
+
+        opt = new Option(DB_HOST_OPTION,
+                         DB_HOST_OPTION_NAME,
+                         true,
+                         "Database hostname used (localhost, myserver) or IP address (127.0.0.1, 192.168.12.1). Cannot be used in conjunction with --" +
```
Comment (mboussaa) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java:496]: add e.g., to (127.0.0.1, 192.168.12.1)
(e.g., 127.0.0.1, 192.168.12.1)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,686 @@
...
+                               URL_OPTION_NAME + ".");
+        opt.setRequired(false);
+        opt.setArgName(DB_HOST_OPTION_NAME.toUpperCase());
+        opt.setArgs(1);
+        options.addOption(opt);
+
+        opt = new Option(DB_PORT_OPTION,
+                         DB_PORT_OPTION_NAME,
+                         true,
+                         "Database port used (8080, 8443, etc...). Will use database vendor default port if not provided. Cannot be used in conjunction with --" +
```
Comment (mboussaa) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java:506]: same use e.g.,

Comment (mboussaa) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java:506]: a sentence cannot start with "Will use"

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,686 @@
...
+                               URL_OPTION_NAME + ".");
+        opt.setRequired(false);
+        opt.setArgName(DB_PORT_OPTION_NAME.toUpperCase());
+        opt.setArgs(1);
+        options.addOption(opt);
+
+        opt = new Option(MYSQL_TIMEZONE_OPTION,
+                         MYSQL_TIMEZONE_OPTION_NAME,
+                         true,
+                         "Specific to MySQL, sets the timezone used by the MySQL server. Default to UTC. Cannot be used in conjunction with --" +
```
Comment (mboussaa) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java:516]: same the sentences are not well formed

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java [hunk 6]
```diff
@@ -101,7 +101,9 @@ public class ChangeHttpConfiguration {
     public static void main(String[] args) {
 
         try {
-            changeHttpConfiguration(args);
+            if (changeHttpConfiguration(args)) {
+                log("Http configuration successfully applied");
```
Comment (mboussaa) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java:105]: I thought we changed this

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java [hunk 7]
```diff
@@ -350,11 +358,13 @@ private static void setCurrentConfiguration(String scheme, String port) {
             ChangeHttpConfiguration.port = isHttps ? WebProperties.WEB_HTTPS_PORT.getValueAsInt()
                                                    : WebProperties.WEB_HTTP_PORT.getValueAsInt();
         }
+        System.out.println("Applying configuration: scheme=" + (isHttps ? "https" : "http") + " hostname=" +
```
Comment (mboussaa) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java:361]: it should be INFO: Applying config...

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java [hunk 8]
```diff
@@ -350,11 +358,13 @@ private static void setCurrentConfiguration(String scheme, String port) {
...
+                           (hostname != null ? hostname : "<unchanged>") + " port=" + ChangeHttpConfiguration.port);
```
Comment (mboussaa) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java:362]: we cannot extract the hostname when hostname is null ?

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java [hunk 9]
```diff
@@ -350,11 +358,13 @@ private static void setCurrentConfiguration(String scheme, String port) {
...
     }
 
     private static String answerAndLog(String line) {
         if (isDebug) {
-            log(" -> Set " + line);
+            log("DEBUG:  -> Set " + line);
```
Comment (mboussaa) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java:367]: maybe the debug is not needed for -> Set, not sure

**Review by mboussaa — APPROVED** [review]
with some minor comments. Code comments and documentation is missing too

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java (hunk 8 above)
Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java:362]: I can try, but it is annoying as it changes the organization of the code. I could also display this message at the end which allows not to change the organization

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java (hunk 9 above)
Comment (mboussaa) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java:367]: I meant the hole block could be a debug line 
maybe not a good idea

Comment (mboussaa) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java:367]: up to you

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java (hunk 1 above)
Comment (nebilbenmabrouk) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java]: Using the same letter for both options is not good for me (even with capital letters).
We can use more than one letter, for example `-pwd` for password

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java [hunk 10]
```diff
@@ -0,0 +1,796 @@
...
+
+    public static final String SPRING_DATASOURCE_PASSWORD = "spring.datasource.password";
+
+    public static final String ALT_SPRING_DATASOURCE_DRIVER_CLASS = "spring.datasource.driver-class-name";
+
+    private static final Map<String, String> configurationFilesPerTarget = new HashMap<>();
+
+    // map of database properties file per component for hsqldb
+    // (contains the user credentials that will be created when the database is first created)
+    private static final Map<String, String> hsqldbConfigurationFilesPerTarget = ImmutableMap.of("rm",
```
Comment (nebilbenmabrouk) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java:126]: why HSQLDB has a different list of target files than the other DBs ?

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java (hunk 1 above)
Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java]: @nebilbenmabrouk say that to the linux developpers, many linux commands have parameters in both lowercase and uppercase.

I think it's perfectly fine to use uppercase letters

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java (hunk 8 above)
Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java]: @mboussaa I've done the requested improvement

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java (hunk 9 above)
Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java:414]: I kept this as it was

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java [hunk 11]
```diff
@@ -0,0 +1,796 @@
...
+                log(vendor + " database configuration successfully applied to " + component + ".");
+            }
+        } catch (ChangeDBConfigurationException e) {
+            System.err.println("ERROR : " + e.getMessage());
+            if (e.getCause() != null) {
+                System.err.println(e.getCause().getMessage());
+            }
+            if (e.getAdditionalInfo() != null) {
+                log(e.getAdditionalInfo());
+            }
```
Comment (nebilbenmabrouk) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java:288]: I suggest to display such error details only in debug mode.

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java [hunk 12]
```diff
@@ -0,0 +1,796 @@
...
+            System.exit(1);
+        } catch (Exception e) {
+            System.err.println("ERROR : " + e.getMessage());
+            if (e.getCause() != null) {
+                System.err.println(e.getCause().getMessage());
+            }
+            e.printStackTrace();
```
Comment (nebilbenmabrouk) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java:295]: Sam here

**Review by nebilbenmabrouk — APPROVED** [review]
LGTM

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java (hunk 11 above)
Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java:288]: I don't see why, even not in debug mode, I want to see the full error details

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java (hunk 2 above)
Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java:415]: I added line breaks as suggested

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java (hunk 3 above)
Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java]: done

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java (hunk 4 above)
Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java]: I did the requested changes

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java (hunk 6 above)
Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeHttpConfiguration.java]: updated

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java (hunk 11 above)
Comment (nebilbenmabrouk) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java:288]: Both cases are correct. In many systems, in the debug/verbose mode we have more detailed outputs or errors

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java (hunk 1 above)
Comment (mboussaa) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ChangeDBConfiguration.java]: I agee with Fabien I just looked at scp command as example, it is the case

**Pull request merged**